packages/eslint-config-vtex/rules/typescript.js

```diff
@@ -167,15 +167,30 @@ module.exports = {
           },
         ],
 
-        // Require camel case names
-        // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/camelcase.md
+        // Require a consistent naming convention
+        // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md
         camelcase: 'off',
-        '@typescript-eslint/camelcase': [
+        '@typescript-eslint/naming-convention': [
           'error',
           {
-            properties: 'never',
-            genericType: 'always',
-            ignoreDestructuring: false,
+            selector: 'default',
+            format: ['camelCase'],
+            leadingUnderscore: 'allow',
+            trailingUnderscore: 'allow',
+          },
+          {
+            selector: 'variable',
+            format: ['camelCase', 'UPPER_CASE', 'PascalCase'],
```

Comment:
Don't know how we were able to create module-scoped variables such as

`const MyComponent = () => ...` without eslint complaining about the casing. I've added `PascalCase` to support this use case.